backend/pong/oauth2/create_oauth2_account.py

```diff
@@ -21,19 +21,19 @@ def create_oauth2_user(
     if not oauth2_user_serializer.is_valid():
         return CreateOAuth2UserResult.error(oauth2_user_serializer.errors)
     player_data: dict = {"display_name": display_name}
-    oauth2_account_result = create_account.create_account(
-        oauth2_user_serializer, player_data
+    # create_accountのdataを取得する前に、UserSerializerを保存する必要がある
+    oauth2_user: models.User = oauth2_user_serializer.save()
+    oauth2_account_result: create_account.CreateAccountResult = (
+        create_account.create_account(oauth2_user_serializer, player_data)
     )
     if not oauth2_account_result.is_ok:
+        oauth2_user.delete()
         return CreateOAuth2UserResult.error(
             oauth2_account_result.unwrap_error()
         )
-    # todo: typecehckingが通らないため、一旦getで取得
-    oauth2_user: models.User = models.User.objects.get(
-        id=oauth2_account_result.unwrap()["id"]
-    )
-    # oauth2_user: models.User = oauth2_account_result.unwrap()
-    return CreateOAuth2UserResult.ok(oauth2_user)
+    oauth2_user_serializer_data: dict = oauth2_user_serializer.data
```

Comment:
[imo]
今の実装だと意味は同じですが、`create_account()` の返り値を使う方が今後の変更時のバグを防げそうなので良いかもです
```suggestion
    oauth2_user_serializer_data: dict = oauth2_account_result.unwrap()
```

Reply:
ありがとうございます！コミットメッセージも変更したいため、https://github.com/42-pong/42-pong/pull/236#discussion_r1938240364 とまとめて修正し、forcepushします!